src/lib.rs

```diff
@@ -45,14 +49,23 @@ pub struct FieldOffset<T, U>(
     ///     of.apply(foo)
     /// }
     /// ```
-    PhantomData<(PhantomContra<T>, U)>,
+    PhantomData<(PhantomContra<T>, U, PinFlag)>,
 );
 
+/// Type that can be used in the `PinFlag` parameter of `FieldOffset` to specify that
+/// This projection is valid on Pin types.
+/// See documentation of `FieldOffset::new_from_offset_pinned`
+pub enum AllowPin {}
+
+/// Type that can be used in the `PinFlag` parameter of `FieldOffset` to specify that
+/// This projection is valid on Pin types.
+pub type NotPinned = ();
```

Comment:
Is there a reason this is a type alias rather than a new type?

Reply:
No, there is no reasons for that. I can change it to a new type.